core/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -175,6 +206,16 @@ public String concreteSingleIndex(ClusterState state, IndicesRequest request) {
     }
 
     /**
+     * @return whether the specified alias or index exists. If the alias or index contains datemath then that is resolved too.
+     */
+    public boolean hasIndexOrAliasAndResolveDateMath(String aliasOrIndex, ClusterState state) {
+        assert expressionResolvers.get(0) instanceof DateMathExpressionResolver;
+        List<String> resolvedDateMath = expressionResolvers.get(0).resolve(new Context(state, IndicesOptions.lenientExpandOpen()), Collections.singletonList(aliasOrIndex));
+        assert resolvedDateMath.size() == 1;
```

Comment:
The `DateMathExpressionResolver` never expands like `WildcardExpressionResolver` does. I added this assertion there for extra protection in the case the `DateMathExpressionResolver` gets changed.

But I see your point, even if that would happen (which is impossible now) it should be dealt with later on.